pipeline/forkhandler.go

```diff
@@ -22,19 +23,20 @@ func NewForkHandle() *ForkHandler {
 func (f *ForkHandler) handleUndo(
 	clock *pbsubstreams.Clock,
 	cursor *bstream.Cursor,
-	moduleOutputCache *outputs.ModulesOutputCache,
-	storeMap map[string]*state.Store,
+	outputsCache execout.CacheEngine,
+	storeAccessor *store.Accessor,
 	respFunc func(resp *pbsubstreams.Response) error,
 ) error {
 	if moduleOutputs, found := f.reversibleOutputs[clock.Number]; found {
 		if err := returnModuleDataOutputs(clock, bstream.StepUndo, cursor, moduleOutputs, respFunc); err != nil {
 			return fmt.Errorf("calling return func when reverting outputs: %w", err)
 		}
 		for _, moduleOutput := range moduleOutputs {
-			if outputCache, ok := moduleOutputCache.OutputCaches[moduleOutput.Name]; ok {
-				outputCache.Delete(clock.Id)
+			outputsCache.UndoBlock(moduleOutput.Name, clock.Id)
```

Comment:
Wouldn't it be useful for the `CacheEngine` to just be instructed to `UndoBlock()` with a block ID ?

What happens if we forget to loop on a certain module (we often check if `isOutput`).. That would ruin the internal stability of the CacheEngine. So it's his responsibility to loop through his modules and make sure he properly rolled out whatever he had. No?